src/Onboarding/Routes/LocationRoute.php

```diff
@@ -0,0 +1,53 @@
+<?php
+
+namespace Give\Onboarding\Routes;
+
+use WP_REST_Request;
+use Give\API\RestRoute;
+use Give\Onboarding\Helpers\FormatList;
+
+class LocationRoute implements RestRoute {
+
+	protected $endpoint = 'onboarding/location';
+
+	public function handleRequest( WP_REST_Request $request ) {
+
+		$countryCode = $request->get_param( 'countryCode' );
+		$statesList  = give_get_states( $countryCode );
+
+		$statesList[''] = '-';
+
+		return [
+			'states' => FormatList::fromKeyValue( $statesList ),
+		];
+	}
+
+	/**
+	 * @inheritDoc
+	 */
+	public function registerRoute() {
+		register_rest_route(
+			'give-api/v2',
+			$this->endpoint,
+			[
+				[
+					'methods'             => 'GET',
+					'callback'            => [ $this, 'handleRequest' ],
+					'permission_callback' => function() {
+						return current_user_can( 'manage_options' );
+					},
+					'args'                => [
+						'countryCode' => [
+							'type'              => 'string',
+							'required'          => true,
+							// 'validate_callback' => [ $this, 'validateSetting' ],
+							'sanitize_callback' => 'sanitize_text_field',
+						],
+					],
+				],
+				// Register our schema callback.
+				// 'schema' => [ $this, 'getSchema' ],
```

Comment:
The utility of implementing schema for Give endpoints feels a little "meh"/not entirely apparent to me, but it is part of the standard for WP REST endpoints, is quick to do, and so far is in place for most other GiveWP REST endpoints, so I think it'd be nice to see it in place here. Not a blocker though.

Reply:
Fair point. It isn't something that I'm familiar with, so I put it off. Thanks for keeping me honest!

Reply:
Resolved by 45550e5c3